test/jackdaw/serdes/fressian_test.clj

```diff
@@ -0,0 +1,48 @@
+(ns jackdaw.serdes.fressian-test
+  (:require [clojure.spec.alpha :as s]
+            [clojure.test :refer :all]
+            [clojure.test.check.clojure-test :refer [defspec]]
+            [clojure.test.check.generators :as gen]
+            [clojure.test.check.properties :as prop]
+            [clojure.data.fressian :as fressian]
+            [jackdaw.serdes.fressian :as jsf])
+  (:import java.net.URI
+           [org.fressian.handlers WriteHandler ReadHandler]))
+
+(defspec fressian-roundtrip-test 20
+  (testing "Fressian data is the same after serialization and deserialization."
+    (prop/for-all [x (gen/fmap #(apply str %) (gen/vector gen/char-alpha 100))]
+                  (is (= x (->> (.serialize (jsf/fressian-serializer) nil x)
+                                (.deserialize (jsf/fressian-deserializer) nil)))))))
+
+(defspec fressian-reverse-roundtrip-test 20
+  (testing "Fressian data is the same after deserialization and serialization."
+    (prop/for-all [x (gen/fmap #(apply str %) (gen/vector gen/char-alpha 100))]
+                  (let [bytes (.serialize (jsf/fressian-serializer) nil x)]
+                    (is (= (seq bytes)
+                           (seq (->> (.deserialize (jsf/fressian-deserializer) nil bytes)
+                                     (.serialize (jsf/fressian-serializer) nil)))))))))
+
+(def uri-tag "jackdaw/uri")
+
+(def write-handlers
+  (-> (merge {URI {uri-tag (reify WriteHandler
+                             (write [_ writer uri]
+                               (.writeTag writer uri-tag 1)
+                               (.writeString writer (str uri))))}}
+             fressian/clojure-write-handlers)
+      fressian/associative-lookup
+      fressian/inheritance-lookup))
+
+(def read-handlers
+  (-> (merge {uri-tag (reify ReadHandler
+                        (read [_ reader tag component-count]
+                          (URI. (.readObject reader))))}
+             fressian/clojure-read-handlers)
+      fressian/associative-lookup))
```

Comment:
Not sure if that’s expected, but shouldn’t the read handler also use `fressian/inheritance-lookup` like the write handler ?

Reply:
@gphilipp I adapted that test from https://github.com/clojure/data.fressian/wiki/Creating-custom-handlers, and Stu didn't include the inheritance-lookup on the read side.  Inheritance is necessary when you've got an in-memory thingy and need to serialize it somehow, and somewhere in its inheritance hierarchy some superclass or interface knows how to be serialized.  It's less important when you've got a serialized thing and want to read it into a specific class, e.g. a `java.net.URI` in this case.